File: 1-c/cash.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    // total cents that will also be updated after subtraction
    int cents;
    // counter for the coins needed
    int n = 0;
    // naming each quantity of cents accordignly
    int quarter = 25;
    int dime = 10;
    int nickel = 5;
    int penny = 1;

    // runs once and prompts the user for a positive int for the cents
    do
    {
        cents = get_int("Change owned: ");
    }
    while (cents < 0);

    // each while checks for the remaining and also increments the number of coins
    while (cents >= quarter)
    {
        cents -= quarter;
        n++;
    }
    while (cents >= dime)
    {
        cents -= dime;
        n++;
    }
    while (cents >= nickel)
    {
        cents -= nickel;
        n++;
    }
    while (cents >= penny)
    {
        cents -= penny;
        n++;
    }

    // prints the number of coins needed for the change
    printf("%i\n", n);
}

File: 6-python/cash.py
from cs50 import get_float

while True:
    dollars = get_float("Change: ")
    if dollars > 0:
        break

n = 0
cents = round(dollars * 100)

while cents >= 25:
    cents -= 25
    n += 1

while cents >= 10:
    cents -= 10
    n += 1

while cents >= 5:
    cents -= 5
    n += 1

while cents >= 1:
    cents -= 1
    n += 1

print(n)
